src/app/ReadHandler.h

```diff
@@ -565,6 +565,8 @@ class ReadHandler : public Messaging::ExchangeDelegate
 
     // TODO (#27675): Merge all observers into one and that one will dispatch the callbacks to the right place.
     Observer * mObserver = nullptr;
+
+    InteractionModelEngine * mpImEngine = nullptr;
```

Comment:
We should be careful to add additional pointer in RaedHandler, which could cause memory pressure in matter devices.

Reply:
I am planning to work on WriteHandler and CommandHandler for the similar behavior as well. Is it going to be a big issue?

Reply:
It's going to be an issue... We're already over memory targets; we need to be reducing memory, not using more.

Reply:
Pointer removed.